Feat/supabase avatar component

HwaEVE opened the pull request.

HwaEVE committed:
refactor: 로그인을 했을 때 유저가 가입한 프로필 사진으로 변경

HwaEVE committed:
refactor: 로딩·에러 처리 로직을 LoadingWrapper로 추출

HwaEVE committed:
chore(ui): 네비게이션 바에 사용자 아바타 팝오버 UI 및 로그아웃 기능 구현

HwaEVE committed:
feat(ui): 아바타 클릭 팝오버 스타일 개선 및 프로필·로그아웃 기능 추가

HwaEVE committed:
chore(ui): 사용하지 않는 selectedTimeframe 상태 훅 유지

HwaEVE committed:
feat(ui): SupabaseListener 등록해 로그인 상태 변화 시 사용자 Post 처리

HwaEVE committed:
fix: SupabaseListener url 변경

HwaEVE committed:
refactor: useEffect 로직을 useUser/useClickOutside hook으로 분리

HwaEVE committed:
refactor: 로그인 아바타 클릭시 팝업을 avatarMenu.tsx를 components 분리 후 오타 수정

HwaEVE committed:
feat: 유저 프로필 구현 및 기자 신청

HwaEVE committed:
feat: avatarMenu.tsx의 파일 위치 변경 및 운영모드 추가

HwaEVE committed:
fix: useUser.ts 사용하지 않는 임포트 제거

HwaEVE committed:
refactor: 프로필 페이지의 기자신청 로직 개선

HwaEVE committed:
refactor: useClickOutside.ts과 useUser.ts의 폴더 이동

HwaEVE committed:
refactor: avatarMenu의 useAdminFlag를 훅으로 refactoring

HwaEVE committed:
refactor: 관리자는 프로필에서 관리자 딱지를 얻도록 수정

HwaEVE committed:
refactor(profile): useComments, usePosts, useReporterApplication 등 커스텀 훅으로 분리

HwaEVE committed:
refactor: main과 ProfilePage에서 기존 네비게이션 코드 삭제하고 NavBar 컴포넌트 사용

HwaEVE committed:
refactor: 기자신청 Hook에 있는 api util로 분리

HwaEVE committed:
refactor: main과 profile의 네비게이션 분리 및 사용자의 권한을 확인하는 기능 수정

yulmucha commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,209 @@
+'use client';
```
이 페이지 컴포넌트 전체가 꼭 클라이언트 컴포넌트여야 하나요?

yulmucha commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,209 @@
+  useEffect(() => {
+    async function fetchComments() {
+      try {
+        const res = await fetch(`${API_BASE}/api/comments`);
+        const data: Comment[] = await res.json();
```
`fetch()` 부분은 인혁님이 하신 것처럼 따로 분리하시면 더 좋을 것 같아요.

yulmucha commented:
이 PR 때문은 아닌 것 같지만, /main 접속하면 다음과 같은 에러가 발생하네요.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/f07d8b07-8741-425a-b3fd-18d76265f82e" />

<img width="1377" alt="image" src="https://github.com/user-attachments/assets/0b5b090b-a352-4dc3-8754-1ba85a9146f0" />

yulmucha commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,209 @@
+      const token = session.access_token;
+
+      // 2) 백엔드에 POST 요청 (Authorization 헤더에 Bearer 토큰)
+      const res = await fetch(`${API_BASE}/reporter-applications`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      });
+
+      if (!res.ok) {
+        const body = await res.text();
+        throw new Error(`신청 실패: ${body || res.statusText}`);
+      }
```
여기 `fetch()` 부분도 따로 분리하시면 더 좋을 것 같아요

HwaEVE committed:
refactor: 토큰에서 session을 분리 및 줄 간격 수정

HwaEVE committed:
feat: 기자의 신청을 승인을 하면 기자딱지를 붙여주는 것이 아닌 기자권한이 있어야 기자딱지를 붙여줌

HwaEVE committed:
refactor: supabase 유틸 줄 간격 조정

HwaEVE committed:
Merge branch 'main' into feat/supabase-avatar-component

HwaEVE committed:
feat: navBar를 레이아웃에서 사용

HwaEVE committed:
refactor: 훅에서 불필요한 코드 반복을 하던걸 제거 및 최적화 작업 진행
- API_Base를 따로 분리 적용
- 여러번 호출하던  “supabase.auth.getSession() → 토큰 추출”을 하나의 훅에서 진행
